workbench/runtime.py

```diff
@@ -297,11 +297,14 @@ def get_real_user(self, _):
 
     def _patch_xblock(self, block):
         """Add required attributes by some legacy XBlocks such as the LTI Consumer XBlock."""
-        block.location = Mock(html_id=Mock(return_value='course-v1:edX+Demo+2020'))
-        block.course_id = block.location.html_id()
-        block.due = datetime.utcnow()
-        block.graceperiod = timedelta(seconds=0)
-        block.category = 'chapter'
+        try:
+            block.location = Mock(html_id=Mock(return_value='course-v1:edX+Demo+2020'))
+            block.course_id = block.location.html_id()
+            block.due = datetime.utcnow()
+            block.graceperiod = timedelta(seconds=0)
+            block.category = 'chapter'
+        except AttributeError:
+            pass
```

Comment:
any logging in case of exceptions ?

Reply:
we are mocking data so if its already present im not sure if we need any logging for that case.

Reply:
We should still log something here so that if the unexpected case does happen, we know about it. As it is now, if some assumption changes elsewhere in the system, we might not see that it broke this part of the system.